src/Reflection/ClassReflection.php

```diff
@@ -683,9 +770,83 @@ public function getProperty(string $propertyName, ClassMemberAccessAnswerer $sco
 		return $this->properties[$key];
 	}
 
+	public function getInstanceProperty(string $propertyName, ClassMemberAccessAnswerer $scope): ExtendedPropertyReflection
+	{
+		if ($this->isEnum()) {
+			return $this->getNativeProperty($propertyName);
+		}
+
+		$key = $propertyName;
+		if ($scope->isInClass()) {
+			$key = sprintf('%s-%s', $key, $scope->getClassReflection()->getCacheKey());
+		}
+
+		if (!isset($this->instanceProperties[$key])) {
+			foreach ($this->propertiesClassReflectionExtensions as $i => $extension) {
+				if ($i > 0 && !$this->allowsDynamicPropertiesExtensions()) {
+					break;
+				}
+
+				if (!$extension->hasProperty($this, $propertyName)) {
+					continue;
+				}
+
+				$property = $this->wrapExtendedProperty($extension->getProperty($this, $propertyName));
+				if ($scope->canReadProperty($property)) {
+					return $this->instanceProperties[$key] = $property;
+				}
+				$this->instanceProperties[$key] = $property;
+			}
+		}
+
+		if (!isset($this->instanceProperties[$key])) {
+			if ($this->requireExtendsPropertiesClassReflectionExtension->hasInstanceProperty($this, $propertyName)) {
+				$property = $this->requireExtendsPropertiesClassReflectionExtension->getInstanceProperty($this, $propertyName);
+				$this->instanceProperties[$key] = $property;
+			}
+		}
+
+		if (!isset($this->instanceProperties[$key])) {
+			throw new MissingPropertyFromReflectionException($this->getName(), $propertyName);
+		}
+
+		return $this->instanceProperties[$key];
+	}
+
+	public function getStaticProperty(string $propertyName, ClassMemberAccessAnswerer $scope): ExtendedPropertyReflection
+	{
+		$key = $propertyName;
+		if ($scope->isInClass()) {
+			$key = sprintf('%s-%s', $key, $scope->getClassReflection()->getCacheKey());
+		}
+
+		if (!isset($this->staticProperties[$key])) {
+			if ($this->getPhpExtension()->hasStaticProperty($this, $propertyName)) {
+				$property = $this->wrapExtendedProperty($this->getPhpExtension()->getStaticProperty($this, $propertyName));
+				if ($scope->canReadProperty($property)) {
+					return $this->staticProperties[$key] = $property;
+				}
+				$this->staticProperties[$key] = $property;
```

Comment:
What if this would not check `Scope::canReadProperty` and would always return?

Reply:
Wouldn't this an issue for `protected/private` static properties ?

This will skip the possible override by requireExtendsPropertiesClassReflectionExtension::getStaticProperty